### DISCUSSION TITLE
Spherical Box Potential implementation: how to proceeed?

### DISCUSSION THREAD

**Discussion by mb6-31ansatz** [pyscf/pyscf#2535] [Q&A]
Hi,

I would like to ask your help for the development of a simple atom or molecule in a **spherical box potential**.
I have searched in many examples, but I wasn't able to find anything and I am not very familiar with the capabilities of Libcint. 

Therefore, I hope someone can give me some practical advice regarding the implementation of this new potential.

Thank you for the attention.

**Comment by susilehtola** [comment]
Please define what you mean.

**Reply by mb6-31ansatz** [reply]
Hi, 
I would like to implement (if it does not exist already) a step potential V, in which, the potential V = 0 inside the spherical box boundaries. On the other hand, the potential V = V_0 outside the spherical box boundaries.
This means that only the atom/molecule wavefunction tail is affected by this potential. Clearly, the smaller the spherical box becomes, an higher percentage of the wavefunction is affected.

**Reply by susilehtola** [reply]
A step-wise potential cannot be faithfully represented by a Gaussian basis, and all results will be heavily contaminated by basis set truncation error; see our recent work in [arXiv:2408.11595](https://arxiv.org/abs/2408.11595) for discussion.

The implementation in the Gaussian basis is possible for a **rectangular box** by computing the integrals in the Cartesian basis where the basis functions factorize. An analogous implementation for an atom in a spherical box is also possible, as this only requires computing a radial integral.

Computing the spherical box potential's matrix elements for a general molecule is bound to be difficult, since the potential has a sharp boundary, which will introduce significant numerical noise.
